[MM-25659] document config settings for permanent delete of channel (#3825)
mattermost/docs: source/administration/config-settings.rst

| This feature's ``config.json`` setting is ``"EnableAPITeamDeletion": false`` with options ``true`` and ``false``.                                                    |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Enable API Channel Deletion

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

> **True**: The ``api/v4/channels/{channelid}?permanent=true`` API endpoint can be called by System Admins or user with appropriate delete permissions to permanently delete a channel.

Reviewer: ```suggestion
**True**: The ``api/v4/channels/{channelid}?permanent=true`` API endpoint can be called by System Admins or users who have appropriate permissions to permanently delete a channel.
```
